bokeh/core/tests/test_properties.py

```diff
@@ -49,9 +46,18 @@ class Foo(HasProps):
         self.assertEqual(f.y, "bar")
 
         without_defaults = f.properties_with_values(include_defaults=False)
-        del without_defaults['z']
         self.assertDictEqual(dict(x=18, y="bar"), without_defaults)
 
+        f.z[0] = 100
+
+        without_defaults = f.properties_with_values(include_defaults=False)
+        self.assertDictEqual(dict(x=18, y="bar", z=[100,2,3]), without_defaults)
+
+        f.zz = {'a': 10}
+
+        without_defaults = f.properties_with_values(include_defaults=False)
+        self.assertDictEqual(dict(x=18, y="bar", z=[100,2,3], zz={'a': 10}), without_defaults)
+
```

Comment:
Ensure that mutating or setting a property container affects reported values correctly